### PR TITLE
fixed find_module error for python 3.7 or newer

### PR DESCRIPTION
**Pull request by zcodes**
when python version is 3.7 or newer, error occured when open python file

```
AttributeError: module 'vim' has no attribute 'find_module'
```
as comment in the code:

* since python 3.3, the `imp.find_module()` is deprecated
* `importlib.util.find_spec()` is new in python 3.4
* in vim source, if compiled vim with python 3.7 or new, the vim
 python 3 interface will use `find_spec` instead of `find_module`
 and `load_module`. (note: this depends on the python which compiled
 with vim, not the python loaded by vim at runtime.)

to sovle this, at first, python version detect maybe not worked since you can compile vim with python 3.7 and load python 3.6 at runtime, or in revese, second, we can detect module vim has `find_spec` or not, but then we should checking if we can import module vim or not ... to making it simple, i just used try..catch here.

### CODE REVIEW

**Comment by StingNevermore** [comment]
it works for me!

VIM - Vi IMproved 8.1 (2018 May 18, compiled Dec 13 2018 12:07:57)
macOS version
Included patches: 1-577
Compiled by Homebrew
Huge version with MacVim GUI.  Features included (+) or not (-):
+acl               +farsi             +mouse_sgr         -tag_any_white
+arabic            +file_in_path      -mouse_sysmouse    -tcl
+autocmd           +find_in_path      +mouse_urxvt       +termguicolors
+autochdir         +float             +mouse_xterm       +terminal
-autoservername    +folding           +multi_byte        +terminfo
+balloon_eval      -footer            +multi_lang        +termresponse
+balloon_eval_term +fork()            -mzscheme          +textobjects
+browse            +fullscreen        +netbeans_intg     +timers
++builtin_terms    -gettext           +num64             +title
+byte_offset       -hangul_input      +odbeditor         +toolbar
+channel           +iconv             +packages          +transparency
+cindent           +insert_expand     +path_extra        +user_commands
+clientserver      +job               +perl              +vartabs
+clipboard         +jumplist          +persistent_undo   +vertsplit
+cmdline_compl     +keymap            +postscript        +virtualedit
+cmdline_hist      +lambda            +printer           +visual
+cmdline_info      +langmap           +profile           +visualextra
+comments          +libcall           -python            +viminfo
+conceal           +linebreak         +python3           +vreplace
+cryptv            +lispindent        +quickfix          +wildignore
+cscope            +listcmds          +reltime           +wildmenu
+cursorbind        +localmap          +rightleft         +windows
+cursorshape       +lua               +ruby              +writebackup
+dialog_con_gui    +menu              +scrollbind        -X11
+diff              +mksession         +signs             -xfontset
+digraphs          +modify_fname      +smartindent       +xim
+dnd               +mouse             +startuptime       -xpm
-ebcdic            +mouseshape        +statusline        -xsmp
+emacs_tags        +mouse_dec         -sun_workshop      -xterm_clipboard
+eval              -mouse_gpm         +syntax            -xterm_save
+ex_extra          -mouse_jsbterm     +tag_binary
+extra_search      +mouse_netterm     +tag_old_static
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/Applications/MacVim.app/Contents/Resources/vim"
Compilation: clang -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_MACVIM -Wall -Wno-unknown-pragmas -pipe  -DMACOS_X -DMACOS_X_DARWIN  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: clang   -L. -L.  -L/usr/local/lib -o Vim -framework Cocoa -framework Carbon       -lm  -lncurses -liconv -framework AppKit  -L/usr/local/opt/lua/lib -llua5.3 -fstack-protector  -L/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE -lperl  -L/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/config-3.7m-darwin -lpython3.7m -framework CoreFoundation  -framework Ruby

Python 3.6.8 :: Anaconda, Inc.

File: pymode/libs/pkg_resources/__init__.py [hunk 1]
```diff
@@ -2170,7 +2170,27 @@ def _handle_ns(packageName, path_item):
...
+        # use `find_spec` instead of `find_module` and `load_module`.
+        # * if it compiled with feature `+python3/dyn`, vim can load a python
+        # dynamically. The loaded python maybe has a different version compared
+        # with the python used when compiling vim.
+        #
+        # As these reason, it's hard to say we should use `find_spec` or
+        # `find_module`, so here use try/except for the sake of clearness of
+        # the logic.
+        #
+        # As `find_module` is derecated for newer python, we try `find_spec`
```
Comment (diraol) [inline pymode/libs/pkg_resources/__init__.py]: Hi @zcodes there is just a little typo here! `derecated` x `deprecated`. =)

**Comment by diraol** [comment]
Also, my mistake.
The correct word is `plethora`, not `pletora`.

File: pymode/libs/pkg_resources/__init__.py (hunk 1 above)
Comment (zcodes) [reply pymode/libs/pkg_resources/__init__.py]: @diraol sorry, it's my mistake

**Comment by zcodes** [comment]
@diraol I fixed the typo error

**Comment by diraol** [comment]
Thanks!!

**Pull request merged**